src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+
+                //Get filtered item based on keys
+                TStructuralType original = null;
+                string errorMessage = string.Empty;
+                string geterrorMessage = string.Empty;
+
+                Dictionary<string, object> keyValues = new Dictionary<string, object>();
+
+                foreach (string key in _keys)
+                {
+                    object value;
+                    changedObj.TryGetPropertyValue(key, out value);
+
+                    if (value != null)
+                    {
+                        keyValues.Add(key, value);
+                    }
+                }
+
+                try
+                {                    
+                    PatchStatus status = PatchHandler.TryGet(keyValues, out original, out geterrorMessage);
+
+                    DeltaDeletedEntityObject<TStructuralType> deletedObj = changedObj as DeltaDeletedEntityObject<TStructuralType>;
+
+                    if (deletedObj != null)
+                    {
+                        operation = DataModificationOperationKind.Delete;
+
+                        changedObj.Patch(original, PatchHandler);
+
+                        if (PatchHandler.TryDelete(keyValues, out errorMessage) != PatchStatus.Success)
+                        {
+                            //Handle Failed Operation - Delete                           
+                            
+                            if(status == PatchStatus.Success)
+                            {
+                                IDeltaSetItem changedObject = HandleFailedOperation(changedObj, operation, original, errorMessage);
+                                deltaSet.Add(changedObject);
+                                continue;
+                            }                            
+                        }
+                                               
+                        deltaSet.Add(deletedObj);
+                    }
+                    else
+                    {
+                        if (status == PatchStatus.NotFound)
+                        {
+                            operation = DataModificationOperationKind.Insert;
+
+                            if(PatchHandler.TryCreate(changedObj, out original, out errorMessage) != PatchStatus.Success)
+                            {
+                                //Handle failed Opreataion - create
+                                IDeltaSetItem changedObject = HandleFailedOperation(changedObj, operation, original, errorMessage);
+                                deltaSet.Add(changedObject);
+                                continue;
+                            }                            
+                        }
+                        else if(status == PatchStatus.Success)
+                        {
+                            operation = DataModificationOperationKind.Update;
+                        }
+                        else
+                        {
```

Comment:
Since you initialized the value of `operation` to `DataModificationOperationKind.Update` at the beginning of the `loop`, that value will be used here. Might that be misleading?

Reply:
THere is no default enum value unfortunately, this we are getting from vocabs (defined in vocabs) and use that